src/Parser.php

```diff
@@ -3200,9 +3201,16 @@ private function parseMemberAccessExpression($expression):MemberAccessExpression
     private function parseScopedPropertyAccessExpression($expression, $fallbackParentNode): ScopedPropertyAccessExpression {
         $scopedPropertyAccessExpression = new ScopedPropertyAccessExpression();
         $scopedPropertyAccessExpression->parent = $expression->parent ?? $fallbackParentNode;
+
         if ($expression instanceof Node) {
             $expression->parent = $scopedPropertyAccessExpression;
-            $scopedPropertyAccessExpression->scopeResolutionQualifier = $expression; // TODO ensure always a Node
+
+            // scopeResolutionQualifier does not accept `Node` but
+            // `Expression|QualifiedName|Token`. I'm not sure if we can depend
+            // on that being the case.
+            //
+            // @phpstan-ignore-next-line 
```

Comment:
```suggestion
            // @phpstan-ignore-next-line
```

Remove trailing whitespace